libcxx/include/tuple

```diff
@@ -524,22 +588,19 @@ class _LIBCUDACXX_TEMPLATE_VIS tuple
                 __tuple_constructible<
                     tuple<_Args...>,
                     typename __make_tuple_types<tuple,
-                             sizeof...(_Args) < sizeof...(_Tp) ?
-                                 sizeof...(_Args) :
-                                 sizeof...(_Tp)>::type
+                            __lesser<sizeof...(_Args), sizeof...(_Tp)>::value
```

Comment:
Is this necessary for some reason, or just a convenience rewrite? The new version will probably be more expensive, because it now has to instantiate a type for every combination of `_Lh` and `_Rh`, while the old one just compared values, so if this isn't a workaround for something, we should probably go back to the original way this was written.

Reply:
It was for convenience, but can be reverted at this point. When I was nursing that `__indexed` type error I was trying to get rid of the ternary to see if it helped.